documentation/release_notes.markdown

```diff
@@ -4,116 +4,104 @@ title: "Puppet Server: Release Notes"
+The `always_retry_plugins` setting also replaces the [`always_cache_features` setting](https://docs.puppet.com/puppet/4.5/reference/configuration.html#alwayscachefeatures), which is now deprecated. If you set `always_cache_features` to true in previous versions of Puppet Server, set `always_retry_plugins` to false.
 
-* [SERVER-96](https://tickets.puppet.com/browse/SERVER-96)
+-   [PUP-5482](https://tickets.puppetlabs.com/browse/PUP-5482)
 
-### Bug fix: Puppet Server correctly parses complex script arguments
+### New feature: Expanded logging for certificate autosigning attempts
 
-In versions 1.x and 2.2.x, Puppet Server would incorrectly parse commands executed by Puppet code that had complex string interpolation. For example, calls to the `generate()` function --- such as `generate('/bin/sh', '-c', "/usr/bin/python -c 'print \"foo\"'")` --- would spawn a Python REPL and consume a JRuby instance without returning anything. Puppet Server 2.3.0 fixes this issue.
+Starting with version 2.4.0, Puppet Server [logs](./configuration.markdown#logging) message and warnings when an autosign command generates STDERR output or returns a non-zero exit code. Server 2.4.0 also logs autosigning attempts at the INFO level, rather than DEBUG, to help make autosigning issues easier to diagnose without changing Server's logging level.
 
-* [SERVER-1160](https://tickets.puppet.com/browse/SERVER-1160)
+-   [SERVER-1187](https://tickets.puppetlabs.com/browse/SERVER-1187)
 
-### Known issues
+### Bug fix: Closed memory leak when restarting Server via SIGHUP
 
-#### Modifications to `bootstrap.cfg` might cause problems during upgrades to 2.3.0
+The versions of Trapperkeeper included with Puppet Server 2.3.x leaked a small amount of memory when [restarting Server with a HUP signal](./restarting.markdown). Trapperkeeper 1.4.0, which is included with Puppet Server 2.4.0, resolves this issue.
 
-If you modified `bootstrap.cfg` (for instance, to [enable or disable the Certificate Authority service](./configuration.markdown#service-bootstrapping)), upgrading to Puppet Server 2.3.0 from earlier versions of Puppet Server might fail. For instance, on Red Hat-family distributions of Linux, you might see a warning during the package update:
+-   [TK-372: Jetty JMX Memory Leak in TK WS J9](https://tickets.puppetlabs.com/browse/TK-372)
 
-```
-2016-03-09 11:48:17,460 ERROR [main] [p.t.internal] Error during app buildup!
-java.lang.RuntimeException: Service ':VersionedCodeService' not found
-```
+### Bug fix: Implement DELETE request handling on the `certificate_reqest` endpoint
 
-To resolve this, add this line to your `bootstrap.cfg`:
+Unlike the Ruby Puppet master, previous versions of Puppet Server didn't handle [DELETE requests to the `certificate_request` endpoint](https://docs.puppet.com/puppet/latest/reference/http_api/http_certificate_status.html#delete). Server 2.4.0 resolves this by handling these requests the same way that the Ruby master does.
```

Comment:
Not sure if it's worth mentioning here that one would have to add an auth.conf rule permitting access to the DELETE method for the "certificate_request" endpoint in order to make use of this feature?  Per the rules in the default auth.conf in the Puppet Server package, all DELETE requests made to the "certificate_request" endpoint would be denied.